bugsnag-plugin-android-ndk/src/main/jni/bugsnag_ndk.c

```diff
@@ -159,6 +159,7 @@ JNIEXPORT void JNICALL Java_com_bugsnag_android_ndk_NativeBridge_install(
   // copy event path to env struct
   const char *event_path = bsg_safe_get_string_utf_chars(env, _event_path);
   if (event_path == NULL) {
+    free(bugsnag_env);
```

Comment:
For cleanup code in C, it's generally safer to use a "poor man's finally" approach using goto. That way, any future checked code that can fail doesn't require you to hunt for and match up every freeable resource and every exit point in the function, and all resource-freeing code exists in one place.

```c
{
    resource *resource_a = NULL;
    resource *resource_b = NULL;
    object *some_object = NULL;

    resource_a = get_resource_a();
    if (resource_a == NULL) {
        goto exit;
    }

    if (!validate_resource(resource_a)) {
        goto exit;
    }

    resource_b = get_resource_b(resource_a);
    if (resource_b == NULL) {
        goto exit;
    }

    if (!validate_resource(resource_b)) {
        goto exit;
    }

    some_object = copy_object_from_resource(resource_b);

exit:
    free(resource_a);
    free(resource_b);
    return some_object;
}
```

Reply:
Refactored to `free` when the allocated struct is not assigned to `bsg_global_env`.